### ISSUE TITLE
[BUG] VerifyVersionConstantsIT is unreliable

### ISSUE THREAD

**Issue by peternied** [opensearch-project/OpenSearch#12407]
### Describe the bug

testLuceneVersionConstant did not detect that when the Lucene version was updated [1] the test was broken 
```
java.lang.AssertionError: 
Expected: <9.10.0>
     but: was <9.9.2>
```

- Root Cause https://github.com/opensearch-project/OpenSearch/pull/12392

### Related component

Build

### To Reproduce

1. Update the lucene version
2. Merge change into main
3. Run the test [org.opensearch.qa.verify_version_constants.VerifyVersionConstantsIT.testLuceneVersionConstant](https://build.ci.opensearch.org/job/gradle-check/34002/testReport/org.opensearch.qa.verify_version_constants/VerifyVersionConstantsIT/testLuceneVersionConstant/)

### Expected behavior

Test should fail when the version is upgraded, should pass all other times.

**Comment by peternied** [comment]
[Triage - attendees [1](https://github.com/peternied) [2](https://github.com/dbwiddis) [3](https://github.com/lukas-vlcek) [4](https://github.com/kotwanikunal) [5](https://github.com/linuxpi)]
Thanks for filing, look forward to seeing this fixed

**Comment by peternied** [comment]
[Triage - attendees [1](https://github.com/peternied) [2](https://github.com/dbwiddis) [3](https://github.com/lukas-vlcek) [4](https://github.com/kotwanikunal) [5](https://github.com/linuxpi)]
Thanks for filing - consider contributing with a pull request

**Comment by andrross** [comment]
We have a number of "qa" tests that introduce cross-branch dependencies and result in temporary breakages until all branches are in sync, so to some extent this is by design. There are a number of issues discussing this problem and potential improvements:

- https://github.com/opensearch-project/OpenSearch/issues/8070
- https://github.com/opensearch-project/OpenSearch/issues/2350
- https://github.com/opensearch-project/OpenSearch/issues/2533

@peternied I'm going to remove the "flaky-test" label as I think the failure here is a result of the design of the test. That's not to say it can't be improved, but I don't think it's quite the same category as other flaky tests. Let me know if you disagree.

**Comment by peternied** [comment]
@andrross I disagree - if we rev the Lucene version the test should immediately break, unless I don't understand how this test is executed. 

Can you help confirm that the boundary of the changing the lucence version show have the rest request/response validation that can be done without requiring a change in another branch or as part of the backward compatibility suite?  If the answer is the boundary is complex, then I'd be curious why this test should cross that boundary.

**Comment by andrross** [comment]
> if we rev the Lucene version the test should immediately break, unless I don't understand how this test is executed.

@peternied Sorry, I think I misunderstood this issue. It works like this: we have the Version.java file that defines a static mapping of OpenSearch version to Lucene version. What this test does is:

```
for each version in `BuildParams.bwcVersion.indexCompatible`:
 - Start an OpenSearch cluster for this version
 - call the `GET /` API and verify its response matches what is defined in the Version.java of the local code. 
```

That means if we change the Lucene version on main (aka 3.0), then the test will still pass because it will use the local code to start the 3.0 cluster. If we change the Lucene version on the 2.x branch, then this test will break on main until we update the 2.x version mapping on main. Does that make sense?

**Comment by andrross** [comment]
The intent of this test, as I understand it, is to make sure the forward branches get updated when an older branch (e.g. 1.x or 2.x) update the version of Lucene that they use.

**Comment by peternied** [comment]
> If we change the Lucene version on the 2.x branch, then this test will break on main until we update the 2.x version mapping on main. Does that make sense?

Thanks for breaking down the mechanics - much appreciated.

Does it make sense that a contribution that was created before an off-branch version change was issued cannot pass CI until it merges from main - I'm having a harder time understanding that choice.

This sounds like rather than having a separate dependency (e.g. another maven package named OpenSearchVersionAllThing) using the multi-branched nature of git is being repurposed to hold this information.  Does that seem like a fair statement?

**Comment by andrross** [comment]
> Does it make sense that a contribution that was created before an off-branch version change was issued cannot pass CI until it merges from main - I'm having a harder time understanding that choice.

This is something that happens quite a lot (see #2350). The fundamental issue is that the thing we're testing compatibility against (e.g. 2.x, the next unreleased minor) is a moving target. The CI failures tell you the code in your PR is not compatible with that next unreleased minor version. I honestly don't have any good ideas for how to detect that the failure is because of the code you introduced versus a separate change you haven't rebased.

**Comment by shwetathareja** [comment]
https://build.ci.opensearch.org/job/gradle-check/42005/ again with assertion failure between 9.11.1 and 9.11.0

**Comment by shwetathareja** [comment]
9.11.1 lucene version bump for 2.16 is merged in main as well.
